From 3f2dda849ebeb6385a1d375bf26a1db096e4c3fb Mon Sep 17 00:00:00 2001
From: sudokamikaze <pulshencode@outlook.com>
Date: Fri, 19 Aug 2016 17:19:31 +0300
Subject: [PATCH] Fix

---
 src/module.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/module.c b/src/module.c
index 5f58620..d54c201 100644
--- a/src/module.c
+++ b/src/module.c
@@ -159,7 +159,7 @@ int module_is_available(char *module_name) {
   char *mod_argv[] = {
     modprobe_bin,
     "--dry-run",
-    "--quiet",sudo
+    "--quiet",
     module_name,
     NULL
   };
